src/vl-map-baselayer.js

```diff
@@ -81,35 +94,31 @@ export class VlMapBaseLayer extends vlElement(HTMLElement) {
     return this._createdVectorSource;
   }
 
-  _configureMap() {
+  async _configureMap() {
     if (this._map) {
-      this._map.addBaseLayerAndOverlayMapLayer(this._createBaseLayer(), this._createBaseLayer());
+      this._map.addBaseLayerAndOverlayMapLayer(await this._createBaseLayer(), await this._createBaseLayer());
+      // nodig anders is map initially blanc
+      this._map.render();
     }
   }
 
-  _createWMTSSource() {
-    const size = OlExtent.getWidth(this._projection.getExtent()) / 256;
-    const resolutions = new Array(16);
-    const matrixIds = new Array(16);
-    for (let z = 0; z < 16; ++z) {
-      resolutions[z] = size / Math.pow(2, z);
-      matrixIds[z] = z;
-    }
+  async _createWMTSSource() {
+    const params = {
+      'request': 'getcapabilities',
+      'service': 'wmts',
+      'version': '1.0.0',
+    };
 
-    return new OlWMTSSource({
-      url: this.url,
+    const urlConstruct = new URL(this.url);
+    urlConstruct.search = new URLSearchParams(params);
+
+    const response = await fetch(urlConstruct.href).then((response) => response.text());
+    const parser = new WMTSCapabilities();
+    const val = optionsFromCapabilities(parser.read(response), {
       layer: this.layer,
       matrixSet: 'BPL72VL',
```

Comment:
deze wijzigingen lijken mij voor een andere feature dan voor "ik wil dynamisch een extent kunnen zetten"? niet beter in 2 PRs te doen dan?

die code hier mag zomaar verdwijnen zonder gevolgen?

Reply:
Het is me ook niet duidelijk waarvoor deze wijzigingen uitgevoerd zijn gezien de extend context. Ik ben voorstander van zo klein mogelijke PR's zodat het helder is wat de context is en bijgevolg de review ook beter uitgevoerd kan worden. Ik stel voor deze wijzigingen in een andere PR uit te voeren.

Reply:
Ik zet dit in een andere PR